edxval/api.py

```diff
@@ -491,34 +492,53 @@ def copy_course_videos(source_course_id, destination_course_id):
     if source_course_id == destination_course_id:
         return
 
-    videos = Video.objects.filter(courses__course_id=unicode(source_course_id))
+    course_videos = CourseVideo.objects.select_related('video', 'video_image').filter(
+        course_id=unicode(source_course_id)
+    )
 
-    for video in videos:
-        CourseVideo.objects.get_or_create(
-            video=video,
+    for course_video in course_videos:
+        dest_course_video, __ = CourseVideo.objects.get_or_create(
+            video=course_video.video,
             course_id=destination_course_id
         )
+        try:
+            VideoImage.create_or_update(
+                course_video=dest_course_video,
+                file_name=course_video.video_image.image.name
+            )
+        except VideoImage.DoesNotExist:
+            pass
```

Comment:
Would we ever get the exception here? I mean, we are using `create_or_update` so if not exist it will create, otherwise update. What is the best approach here?